### PR TITLE
Add CompiledProgram doc

### PR DESCRIPTION
**Pull request by chengduoZH**
# Executor:
image

# CompiledProgram
image
image

# ParallelExecutor
image
image

image

### CODE REVIEW

**Review by JiabinYang — APPROVED** [review]
LGTM

File: doc/fluid/api_guides/low_level/compiled_program_en.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+..  _api_guide_compiled_program_en:
+
+################
+CompiledProgram
+################
+
+The :ref:`api_fluid_CompiledProgram` is used to transform a program for various optimizations. For example, you can call :code:`with_data_parallel` to transform a single card program to multi cards program, and then you can the program with multi cards to run the program.
```
Comment (shanyi15) [inline doc/fluid/api_guides/low_level/compiled_program_en.rst]: and then you can (what) the program with...

File: doc/fluid/api_guides/low_level/compiled_program_en.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    
+    # First create the Executor.
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+
+    # Run the startup program once and only once.
+    exe.run(fluid.default_startup_program())
+    
+    # Run the main program directly without compile.
+    loss, = exe.run(fluid.default_main_program(),
```
Comment (shanyi15) [inline doc/fluid/api_guides/low_level/compiled_program_en.rst]: 这里只fetch了一个，如果加逗号的话，会不会让用户觉得奇怪呢？

File: doc/fluid/api_guides/low_level/executor_en.rst [hunk 3]
```diff
@@ -5,12 +5,31 @@ Executor
 ################
 
 :code:`Executor` realizes a simple executor in which all operators will be executed in order. You can run :code:`Executor` in a Python script. There are two kinds of executors in PaddlePaddle Fluid. One is single-thread executor which is the default option for :code:`Executor` 
-and another is the parallel executor which is illustrated in :ref:`api_guide_parallel_executor_en` .
+and another is the parallel executor which is illustrated in :ref:`api_guide_parallel_executor_en` . The config of `Executor` and :ref:`api_guide_parallel_executor_en` is different, it may be a bit confusing for some users. To make the executor more facility, we introduce :ref:`api_guide_compiled_program_en` , which is unified `Executor` and :ref:`api_guide_parallel_executor_en`.
```
Comment (shanyi15) [inline doc/fluid/api_guides/low_level/executor_en.rst]: one...the other...?

Comment (shanyi15) [reply doc/fluid/api_guides/low_level/executor_en.rst]: which unifies?

File: doc/fluid/user_guides/howto/training/single_node_en.rst [hunk 4]
```diff
@@ -77,19 +77,25 @@ Notes:
...
-2. :code:`ParallelExecutor` should be indicated whether to use CUDA to train. In the mode of graphic card training, all graphic cards will be occupied. Users can configure `CUDA_VISIBLE_DEVICES <http://www.acceleware.com/blog/cudavisibledevices-masking-gpus>`_ to change graphics cards that are being used.
+1. The constructor of :ref:`api_fluid_CompiledProgram` needs to be set with :code:`fluid.Program` to be run which can not be modified at runtime. 
+2. If :code:`exe` is initialized with CUDAPlace, the model will be run in GPU. In the mode of graphics card training, all graphics card will be occupied. Users can configure `CUDA_VISIBLE_DEVICES <http://www.acceleware.com/blog/cudavisibledevices-masking-gpus>`_ to change graphics cards that are being used. 
+3. If :code:`exe` is initialized with CPUPlace, the model will be run in CPU. In this situation, there will be multi-threads are used to run the model, and the threads number are equal to the number of logic cores. Users can configure `CPU_NUM`  to change the number of threads that are being used.
```
Comment (shanyi15) [inline doc/fluid/user_guides/howto/training/single_node_en.rst]: there will be multi-threads are used to run the model 略别扭
User 泛指即可，可以不加s

File: doc/fluid/api_guides/low_level/compiled_program_en.rst (hunk 1 above)
Comment (chengduoZH) [reply doc/fluid/api_guides/low_level/compiled_program_en.rst]: Done

File: doc/fluid/api_guides/low_level/compiled_program_en.rst (hunk 2 above)
Comment (chengduoZH) [reply doc/fluid/api_guides/low_level/compiled_program_en.rst]: Done

File: doc/fluid/api_guides/low_level/executor_en.rst (hunk 3 above)
Comment (chengduoZH) [reply doc/fluid/api_guides/low_level/executor_en.rst]: Done

File: doc/fluid/user_guides/howto/training/single_node_en.rst (hunk 4 above)
Comment (chengduoZH) [reply doc/fluid/user_guides/howto/training/single_node_en.rst]: Done

**Review by shanyi15 — APPROVED** [review]

**Pull request merged**